### ISSUE TITLE
Clients fail to join in time

### ISSUE THREAD

**Issue by nitkul** [kengz/aiva#37]
I ran 'npm run setup --debug' on my Dell and, like many others here, it would exit as "clients" would fail to join in time and then **hang**. Increasing the timeout in `src/start-io.js` to 150000 ms also didn't help, so  I reduced the client count to a custom 2. It doesn't throw errors, but it still **hangs**. Any help?

**Comment by znat** [comment]
Do you have another error in the logs, like a module missing? Are you running it on Python 3?

**Comment by nitkul** [comment]
@znat initially it showed 'autocorrect' missing in 'convo_classifiers.py' but I removed the import statement to make it run for a change

**Comment by znat** [comment]
That's the error I had too. Once fixed (install the package) it worked, all the clients were connected in time

**Comment by nitkul** [comment]
@znat where can I download it from?

**Comment by znat** [comment]
pip install autocorrect

**Comment by nitkul** [comment]
@znat that failed to work :-1:

**Comment by kengz** [comment]
Try to start the IO components individually:
in on terminal window run this:
```shell
npm run server
```

in another, run this:
```
python3 ./lib/client.py
```

See what the outputs are?

A known issue is from the `websocket`, if that's the case try a reinstall
```
pip3 uninstall websocket
pip3 install websocket
```

**Comment by nitkul** [comment]
@kengz npm run still hung my machine but running client.py showed:

`An incompatible websocket library is conflicting with the one we need.
You can remove the incompatible library and install the correct one
by running the following commands:

yes | pip uninstall websocket websocket-client
pip install -U websocket-client`

but i had it already installed!

**Comment by kengz** [comment]
Yep that's a known problem with that python dependency. Could you do an uninstall and reinstall for it as shown above? Then retry - that should fix it

**Comment by nitkul** [comment]
I successfully uninstalled it but when I install `websocket` it says: 'Requirement already up-to-date' :-/

**Comment by kengz** [comment]
python3? just to be absolute do it in py3 (since aiva uses py3)
```shell
python3 -m pip uninstall websocket
python3 -m pip install websocket
```
